lib/medtronic600/NGPHistoryParser.js

```diff
@@ -545,25 +761,31 @@ class NGPHistoryParser {
     }
   }
 
-  * eventsOfType(eventType) {
+  * eventsOfType(...eventTypes) {
     for (const event of this.events) {
-      if (event.eventType === eventType) {
+      if (_.includes(eventTypes, event.eventType)) {
         yield event;
       }
     }
   }
 
-  findSuppressedBasal(tempBasal) {
-    let tempBasalIndex = 0;
-    // TODO - change *eventsOfType to return .entries() so that we already have the index.
-    for (const [index, event] of this.events.entries()) {
-      if (event === tempBasal) {
-        tempBasalIndex = index;
-        break;
+  * eventsOfTypeReverse(...eventTypes) {
+    for (let i = this.events.length - 1; i >= 0; i--) {
+      const event = this.events[i];
+      if (_.includes(eventTypes, event.eventType)) {
+        yield event;
       }
     }
+  }
+
+  // This is here because a Temp Basal with a percentage requires the suppressed basal
+  // to determine the rate. Though this could also be done in the simulator, we do it here
+  // because the event is incomplete without the rate.
+  findSuppressedBasal(tempBasal) {
```

Comment:
Yes, I've confirmed that basal start events always occur after a temp basal has completed, or in the middle of a temp basal if the schedule changes.